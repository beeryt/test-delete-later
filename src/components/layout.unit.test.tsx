import React from 'react'
import { render, waitFor } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

import Layout from './layout'

describe('Layout', () => {
  const defaultMetadata = {
    title: 'Default Title',
    description: 'Default Description',
    siteUrl: 'https://example.com'
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return { site: { siteMetadata: defaultMetadata } }
    })
  })

  afterAll(() => useStaticQuery.mockReset())

  it('renders correctly', async () => {
    render(<Layout />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() => expect(document.title).toBe(defaultMetadata.title))
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })

  it('correctly sets canonical link', async () => {
    const location = { pathname: '/blog/one/' }
    render(<Layout location={location} />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() => expect(document.head).toMatchSnapshot())
    await waitFor(() =>
      expect(
        document.head
          .querySelector('link[rel=canonical]')
          ?.attributes.getNamedItem('href')?.value
      ).toBe(defaultMetadata.siteUrl + location.pathname)
    )
  })
})
