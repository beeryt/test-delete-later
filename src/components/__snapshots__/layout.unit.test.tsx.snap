// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout correctly sets canonical link 1`] = `
<body>
  <div>
    <header>
      <nav
        aria-label="main navigation"
        class="navbar is-primary"
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item is-size-3 has-text-dark"
            href="/"
          >
            Default Title
          </a>
          <div
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger "
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </div>
        </div>
        <div
          class="navbar-menu "
        >
          <div
            class="navbar-end"
          />
        </div>
      </nav>
    </header>
    <main
      id="full-body-wrapper"
    />
    <footer
      class="footer"
    />
  </div>
</body>
`;

exports[`Layout correctly sets canonical link 2`] = `
<head>
  <title>
    Default Title
  </title>
</head>
`;

exports[`Layout renders correctly 1`] = `
<body>
  <div>
    <header>
      <nav
        aria-label="main navigation"
        class="navbar is-primary"
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item is-size-3 has-text-dark"
            href="/"
          >
            Default Title
          </a>
          <div
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger "
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </div>
        </div>
        <div
          class="navbar-menu "
        >
          <div
            class="navbar-end"
          />
        </div>
      </nav>
    </header>
    <main
      id="full-body-wrapper"
    />
    <footer
      class="footer"
    />
  </div>
</body>
`;

exports[`Layout renders correctly 2`] = `
<head>
  <title>
    Default Title
  </title>
  <meta
    content="Default Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;
